AmplifyPlugins/Storage/AWSS3StoragePlugin/Operation/AWSS3StorageDownloadDataOperation.swift

```diff
@@ -82,29 +85,19 @@ public class AWSS3StorageDownloadDataOperation: AmplifyInProcessReportingOperati
             return
         }
 
-        let identityIdResult = authService.getIdentityId()
-
-        guard case let .success(identityId) = identityIdResult else {
-            if case let .failure(error) = identityIdResult {
-                dispatch(StorageError.authError(error.errorDescription, error.recoverySuggestion))
+        let prefixResolver = (storageConfiguration.prefixResolver != nil) ? storageConfiguration.prefixResolver! :
+            StorageAccessLevelAwarePrefixResolver(authService: authService)
+        let prefix = prefixResolver.resolvePrefix(for: request.options.accessLevel,
+                                                  targetIdentityId: request.options.targetIdentityId)
+        switch prefix {
```

Comment:
nit: Some code changes here and below.

```suggestion
        let prefixResolver = storageConfiguration.prefixResolver ??
            StorageAccessLevelAwarePrefixResolver(authService: authService)
        let prefixResolution = prefixResolver.resolvePrefix(for: request.options.accessLevel,
                                                            targetIdentityId: request.options.targetIdentityId)
        switch prefixResolution {
```

Reply:
thanks! updated all operations with this